libexec/bats-preprocess

```diff
@@ -21,27 +22,29 @@ encode_name() {
       elif [[ "$char" =~ [[:alnum:]] ]]; then
         result+="$char"
       else
-        result+="$(printf -- "-%02x" \'"$char")"
+        printf -v 'hex_code' -- "-%02x" \'"$char"
+        result+="$hex_code"
       fi
     done
   fi
 
-  echo "$result"
+  printf -v "$2" '%s' "$result"
 }
 
 tests=()
 index=0
-pattern='^ *@test  *([^ ].*)  *\{ *(.*)$'
+pattern='^ *@test +(.+) +\{ *(.*)$'
 
 while IFS= read -r line; do
+  line="${line//$'\r'}"
   let index+=1
   if [[ "$line" =~ $pattern ]]; then
-    quoted_name="${BASH_REMATCH[1]}"
+    name="${BASH_REMATCH[1]#[\'\"]}"
+    name="${name%[\'\"]}"
     body="${BASH_REMATCH[2]}"
-    name="$(eval echo "$quoted_name")"
```

Comment:
doesn't removing eval, line 41 on `$quoted_name`, remove expected behavior on variable expansion in the string or such?